src/main/java/com/plugish/woominecraft/WooMinecraft.java

```diff
@@ -302,4 +326,53 @@ private void wmc_log(String message, Integer level) {
 				break;
 		}
 	}
+	//Mojang api check
+	private boolean AuCh(Player p) {
+		//check if server is in online/offline mode
+		if (Bukkit.getServer().getOnlineMode()) {
+			return true;
+			//check if the server is connected to a bungee network
+		} else if (Bukkit.spigot().getConfig().getBoolean("settings.bungeecord")) {
+			if (!Players.contains(p.getName() + ':' + p.getUniqueId() + ':' + true)) {
+				if (Players.contains(p.getName() + ':' + p.getUniqueId() + ':' + false)) {
+					p.sendMessage("Mojang Auth: Please Speak with a admin about your purchase");
+					wmc_log("Offline mode not supported");
+					return false;
+				}
+				Bukkit.getScheduler().runTaskAsynchronously(WooMinecraft.instance, () -> {
+					try (InputStream inputStream = new URL("https://api.mojang.com/users/profiles/minecraft/" + p.getName()).openStream(); Scanner scanner = new Scanner(inputStream)) {
+						//if User doesn't exist throws IOException
+						String a = scanner.next();
+						if (isDebug()) {
+							wmc_log(inputStream.toString());
+							wmc_log(a);
+						}
+						if (a.contains(p.getName())) {
+							if (a.contains(p.getUniqueId().toString())) {
+								Players.add(p.getName() + ':' + p.getUniqueId() + ':' + true);
+							} else {
+								//if Username exists but is using the offline uuid(doesn't match mojang records) throw IOException and add player to the list as cracked
+								Players.add(p.getName() + ':' + p.getUniqueId() + ':' + false);
+								throw new IOException("Mojang Auth: PlayerName doesn't match uuid for account");
+							}
+						} else {
+							//add username to the Players list, but as cracked, throws IOE
+							Players.add(p.getName() + ':' + p.getUniqueId() + ':' + false);
+							throw new IOException("Mojang Auth: PlayerName doesn't exist");
+						}
+					} catch (IOException e) {
+						wmc_log(e.getMessage(), 3);
+						p.sendMessage("Mojang Auth:Please Speak with a admin about your purchase");
+						if (isDebug()) {
+							wmc_log(Players.toString());
+						}
+					}
+				});
+			}
+		} else {
+			wmc_log("Server in offline Mode");

```

Comment:
Move this up and return early instead of having all this code in a huge if block.